(torch/elastic) skip logging structured error info if error_file is not set

kiukchung opened the pull request:
Summary:
resolves https://github.com/pytorch/pytorch/issues/73465

This `log.error` is not necessary (and its also not human-friendly formatted) because we end up re-raising the same exception after recording the exception into an error_file (if present). Eventually python should handle this error the way it handles any other errors and will write the trace info into the console. This additional logging produces duplicate error console prints, which affects all users whose schedulers do not set `TORCHELASTIC_ERROR_FILE` env var when calling `torch.distributed.run`.

Test Plan:
Induce an error on the agent process by `kill -15 $AGENT_PID`
```
python -m torch.distributed.run \
   --nproc_per_node 2 \
   --nnodes 1:1 \
   --rdzv_backend c10d \
  --rdzv_endpoint localhost:29500 \
  --monitor_interval 3 test.py
```

Produces

{F704936697}

In contrast to the duplicated error before:

{F704936729}

Differential Revision: D34501852

d4l3k commented on torch/distributed/elastic/multiprocessing/errors/error_handler.py:
```diff
@@ -83,7 +81,20 @@ def record_exception(self, e: BaseException) -> None:
         """
-        _write_error(e, self._get_error_file_path())
+
+        file = self._get_error_file_path()
```
why inline this? isn't this the exact same logic as in `_write_error`?

kiukchung replied:
to reduce complexity. `_write_error()` was a single line function and it wasn't being used anywhere else.

kiukchung replied:
> why inline this? isn't this the exact same logic as in `_write_error`?

oh looks like I published the wrong version, i did actually remove `def _write_error()`

Commit:
(torch/elastic) skip logging structured error info if error_file is not set (#73477)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/73477

resolves https://github.com/pytorch/pytorch/issues/73465

This `log.error` is not necessary (and its also not human-friendly formatted) because we end up re-raising the same exception after recording the exception into an error_file (if present). Eventually python should handle this error the way it handles any other errors and will write the trace info into the console. This additional logging produces duplicate error console prints, which affects all users whose schedulers do not set `TORCHELASTIC_ERROR_FILE` env var when calling `torch.distributed.run`.

Test Plan:
Induce an error on the agent process by `kill -15 $AGENT_PID`
```
python -m torch.distributed.run \
   --nproc_per_node 2 \
   --nnodes 1:1 \
   --rdzv_backend c10d \
  --rdzv_endpoint localhost:29500 \
  --monitor_interval 3 test.py
```

Produces

{F704936697}

In contrast to the duplicated error before:

{F704936729}

Reviewed By: d4l3k

Differential Revision: D34501852

fbshipit-source-id: 5112306ef2d242cc95330a28dc5de7a23a9c0b0d
